sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -879,7 +879,11 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       // columns. Here we Lowercase the column names before passing the partition spec to Hive
       // client, to satisfy Hive.
       // scalastyle:off caselocale
-      orderedPartitionSpec.put(colName.toLowerCase, partition(colName))
+      partition.keys.foreach { partColName =>
+        if (partColName.toLowerCase == colName.toLowerCase()) {
```

Comment:
Ur, according to the PR title, you wanted `Resolving the partition column name based on the resolver`.
But, this is completely ignore the `resolver` setting. Let's be clear what you need be consistent.

Reply:
very valid question, the problem here is the partition spec  string wont match with the column name coming from the table metadata, so when we try to get the partition value detail from TablePartitionSpec based on table metadata column name  we get empty .this is why i am trying to get the partition value based on the column name specified in spec.

Reply:
> I left a few comments, @sujith71955 . The UT is the most important comment. Could you update your PR, please?

I updated my PR based on above comments. thanks all for the valuable time and great suggestions.

Reply:
Is this part not related to the issue in a LOAD command? If so, I thinks it would be better to split this fix into two prs.

Reply:
I checked the code around and I think this is a bug caused by column name resolution in https://github.com/apache/spark/pull/24903/files#diff-159191585e10542f013cb3a714f26075L882. Could you make a pr for this issue only?